Lib/test/test_import/__init__.py

```diff
@@ -877,6 +872,14 @@ def test_pyc_always_writable(self):
             m = __import__(name)
             self.assertEqual(m.x, 'rewritten')
 
+    def test_double_const(self):
+        # Importing double_const checks that float constants
+        # serialiazed by marshal as PYC files don't lose precision
+        # (SF bug 422177).
+        from test.test_import.data import double_const
+        unload('test.test_import.data.double_const')
+        from test.test_import.data import double_const
```

Comment:
Why the double import? Are you concerned the module was previously imported? If so then it's still okay as it would suggest it succeeded previously.

Reply:
Check my previous comments. The purpose of the test is to check that constants are properly serialized/deserialized by marshal when loading code from a PYC cache file.